e2e/test/scenarios/embedding/embedding-dashboard.cy.spec.js

```diff
@@ -423,6 +426,94 @@ describe("scenarios > embedding > dashboard parameters with defaults", () => {
   });
 });
 
+describeEE("scenarios > embedding > dashboard appearance", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+    setTokenFeatures("all");
+  });
+
+  it("should not rerender the static embed preview unnecessarily (metabase#38271)", () => {
+    const dashboardDetails = {
+      name: "dashboard name",
+      enable_embedding: true,
+    };
+
+    const questionDetails = {
+      name: "Orders",
+      query: {
+        "source-table": ORDERS_ID,
+      },
+    };
+    cy.createQuestionAndDashboard({
+      questionDetails,
+      dashboardDetails,
+    }).then(({ body: { dashboard_id } }) => {
+      visitDashboard(dashboard_id);
+    });
+
+    cy.intercept(
+      "GET",
+      "api/embed/dashboard/*",
+      cy.spy().as("embedPreviewSpy"),
+    ).as("embedPreview");
+
+    openStaticEmbeddingModal({
+      activeTab: "parameters",
+      previewMode: "preview",
+      // EE users don't have to accept terms
+      acceptTerms: false,
+    });
+
+    cy.wait("@embedPreview");
+
+    modal().within(() => {
+      cy.findByRole("tab", { name: "Appearance" }).click();
+      cy.get("@embedPreviewSpy").should("have.callCount", 1);
+
+      cy.log("Assert dashboard theme");
+      getIframeBody()
+        .findByTestId("embed-frame")
+        .should("not.have.class", "Theme--transparent");
+      // We're getting an input element which is 0x0 in size
+      cy.findByLabelText("Transparent").click({ force: true });
+      getIframeBody()
+        .findByTestId("embed-frame")
+        .should("have.class", "Theme--transparent");
+      cy.get("@embedPreviewSpy").should("have.callCount", 1);
+
+      cy.log("Assert dashboard title");
+      getIframeBody().findByText(dashboardDetails.name).should("exist");
+      // We're getting an input element which is 0x0 in size
+      cy.findByLabelText("Dashboard title").click({ force: true });
```

Comment:
Instead of finding the element with that label, can we click the label?

Reply:
That's a good point. Do you have an example where we click the labels? I looked up testing-library, and they could only get the input using various matchers e.g. `byPlaceholderText`, `byLabelText`, but to find labels I could use `findByText` which is less specific, or `findByRole` with a lot of hoops. I'd rather use `findByLabelText` with `force: true` because I think it's still more readable.

What do you think?